src/main/scala/com/microsoft/hyperspace/index/plans/logical/BucketUnion.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.plans.logical
+
+import org.apache.spark.sql.catalyst.catalog.BucketSpec
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+/**
+ * [[BucketUnion]] is logical plan for Bucket-aware(partition-aware) Union operation
```

Comment:
nit: What do you mean by Bucket-aware(partition-aware)?

Reply:
I mean Union operation considering bucket specification of input RDDs; Union operation per bucket(partition).

Reply:
Sorry, I should have clarified. I understood that part. Are you using bucket and partition interchageably?

Reply:
I thought they are almost same and used interchangeably in Spark. Let me just remove partition things in the comments.